src/index.ts

```diff
@@ -42,16 +42,21 @@ export const validator = async (token: string, accountSid: string, authToken: st
       reject('Unauthorized: AccountSid or AuthToken was not provided');
       return;
     }
+    
+    // detects if authToken is an object with attributes key and secret to assign hasAPIKey a boolean 
+    const hasAPIKey = !!apikeys ? (!!apikeys.key ? (!!apikeys.secret ? true : false ) : false) : false;
```

Comment:
Can you simplify this logic? Use of double nots and nested ternary is difficult to understand.

Reply:
Where is `apiKeys` variable coming from?

Reply:
Done!  

**change summary**: 
Ternary logic simplified to `const hasAPIKey = credential && typeof credential === 'object' && credential.secret && credential.key;`

Changed the `apiKey` and `authToken` name to credential  gave it a union type  `AuthToken: string | APIKey: {secret: string, key: string}` which resolves to a buffer string when assigned to  the `authorization` const.